tools/serverpod_cli/test/analyzer/models/stateful_analyzer/model_validation/field/default/string/string_bool_default_model_test.dart

```diff
@@ -0,0 +1,152 @@
+import 'package:serverpod_cli/src/test_util/builders/generator_config_builder.dart';
+import 'package:serverpod_cli/src/test_util/builders/model_source_builder.dart';
+import 'package:test/test.dart';
+
+void main() {
+  var config = GeneratorConfigBuilder().build();
+
+  group('Given a class with fields with a "defaultModel" keyword', () {
+    test(
+      'when the field is of type String and the default is set to "This is a default model value", then the field should have a "default model" value',
+      () {
+        var models = [
+          ModelSourceBuilder().withYaml(
+            '''
+          class: Example
+          table: example
+          fields:
+            stringType: String, defaultModel='This is a default model value'
+          ''',
+          ).build()
+        ];
+
+        var collector = CodeGenerationCollector();
+        var definitions =
+            StatefulAnalyzer(config, models, onErrorsCollector(collector))
+                .validateAll();
+
+        expect(collector.errors, isEmpty);
+
+        var definition = definitions.first as ClassDefinition;
+        expect(definition.fields.last.defaultModelValue,
+            'This is a default model value');
+      },
+    );
+
+    test(
+      'when the field is of type String and the default is set to "Another default model value", then the field should have a "default model" value',
+      () {
+        var models = [
+          ModelSourceBuilder().withYaml(
+            '''
+          class: Example
+          table: example
+          fields:
+            stringType: String, defaultModel='Another default model value'
+          ''',
+          ).build()
+        ];
+
+        var collector = CodeGenerationCollector();
+        var definitions =
+            StatefulAnalyzer(config, models, onErrorsCollector(collector))
+                .validateAll();
+
+        expect(collector.errors, isEmpty);
+
+        var definition = definitions.first as ClassDefinition;
+
+        expect(definition.fields.last.defaultModelValue,
+            'Another default model value');
+      },
+    );
+
+    test(
+      'when the field is of type String with an invalid defaultModel value without quotes, then an error is generated',
+      () {
+        var models = [
+          ModelSourceBuilder().withYaml(
+            '''
+        class: Example
+        table: example
+        fields:
+          stringInvalid: String?, defaultModel=InvalidValue
+        ''',
+          ).build()
+        ];
+
+        var collector = CodeGenerationCollector();
+        StatefulAnalyzer(config, models, onErrorsCollector(collector))
+            .validateAll();
+
+        expect(collector.errors, isNotEmpty);
+
+        var firstError = collector.errors.first as SourceSpanSeverityException;
+        expect(
+          firstError.message,
+          'The "defaultModel" value must be a valid string enclosed in single quotes.',
+        );
```

Comment:
Question: This doesn't seem to match the commit i saw previosly for how the message was formatted. The commits might be out of order. There should still be an example of a correctly formatted string in this message.

Reply:
This is already update!